src/Services/src/CmdProcess/ICmdProcessService.cs

```diff
@@ -1,9 +1,10 @@
-﻿using static Tanzu.Toolkit.Services.OutputHandler.OutputHandler;
+﻿using System.Collections.Generic;
+using static Tanzu.Toolkit.Services.OutputHandler.OutputHandler;
 
 namespace Tanzu.Toolkit.Services.CmdProcess
 {
     public interface ICmdProcessService
     {
-        CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, StdOutDelegate stdOutDelegate, StdErrDelegate stdErrDelegate);
+        CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, StdOutDelegate stdOutDelegate, StdErrDelegate stdErrDelegate, Dictionary<string, string> envVars = null);
```

Comment:
I think the order of params might be better as:
* exe
* args
* workdir
* env
* stdout
* stderr

The reason being that the "input" vars would be the leading params and the "output" params would be the trailing params.

Reply:
Fair point - happy to change the order but generally I like to have optional params come after required ones...
How's this? (made the delegates optional as well so they can come last)
`CmdResult RunCommand(string executableFilePath, string arguments, string workingDir, Dictionary<string, string> envVars = null, StdOutDelegate stdOutDelegate = null, StdErrDelegate stdErrDelegate = null);
`